cmd/mimir/main.go

```diff
@@ -53,21 +53,27 @@ const (
 var testMode = false
 
 type mainFlags struct {
-	ballastBytes         int  `category:"advanced"`
-	mutexProfileFraction int  `category:"advanced"`
-	blockProfileRate     int  `category:"advanced"`
-	useBufferedLogger    bool `category:"advanced"`
-	printVersion         bool
-	printModules         bool
-	printHelp            bool
-	printHelpAll         bool
+	ballastBytes                  int     `category:"advanced"`
+	mutexProfileFraction          int     `category:"advanced"`
+	blockProfileRate              int     `category:"advanced"`
+	useBufferedLogger             bool    `category:"advanced"`
+	rateLimitedLogsEnabled        bool    `category:"experimental"`
+	rateLimitedLogsPerSecond      float64 `category:"experimental"`
+	rateLimitedLogsPerSecondBurst int     `category:"experimental"`
+	printVersion                  bool
+	printModules                  bool
+	printHelp                     bool
+	printHelpAll                  bool
 }
 
 func (mf *mainFlags) registerFlags(fs *flag.FlagSet) {
 	fs.IntVar(&mf.ballastBytes, "mem-ballast-size-bytes", 0, "Size of memory ballast to allocate.")
 	fs.IntVar(&mf.mutexProfileFraction, "debug.mutex-profile-fraction", 0, "Fraction of mutex contention events that are reported in the mutex profile. On average 1/rate events are reported. 0 to disable.")
 	fs.IntVar(&mf.blockProfileRate, "debug.block-profile-rate", 0, "Fraction of goroutine blocking events that are reported in the blocking profile. 1 to include every blocking event in the profile, 0 to disable.")
 	fs.BoolVar(&mf.useBufferedLogger, "log.buffered", false, "Use a buffered logger to reduce write contention.")
+	fs.BoolVar(&mf.rateLimitedLogsEnabled, "log.rate-limit-enabled", false, "Use rate limited logger to reduce the number of logged messages per second.")
+	fs.Float64Var(&mf.rateLimitedLogsPerSecond, "log.rate-limit-logs-per-second", 10000, "Maximum number of messages per second to be logged. When 0 there is no rate limit.")
```

Comment:
I don't think it's true that 0 doesn't rate limit. The rate limited logger just accepts whatever value we pass and a value of 0 means to just use the burst. But saying "0 = no rate limit" may be misleading here, so I would just remove it, given we expect the user to simply use `-log.rate-limit-enabled` to enable / disable it.

```suggestion
	fs.Float64Var(&mf.rateLimitedLogsPerSecond, "log.rate-limit-logs-per-second", 10000, "Maximum number of messages per second to be logged.")
```